sdk/reservations/Azure.ResourceManager.Reservations/tests/Tests/PurchaseReservationOrderTests.cs

```diff
@@ -0,0 +1,124 @@
+            }
+        }
+
+        [TestCase]
+        [RecordedTest]
+        public async Task TestPurchaseReservationOrderSharedScopeMonthly()
+        {
+            var purchaseRequestContent = CreatePurchaseRequestContent("Shared", "Monthly");
+            var response = await Tenant.CalculateReservationOrderAsync(purchaseRequestContent);
+            var purchaseResponse = await collection.CreateOrUpdateAsync(WaitUntil.Completed, response.Value.Properties.ReservationOrderId, purchaseRequestContent);
+
+            await purchaseResponse.WaitForCompletionAsync();
+            validateTestResult(purchaseResponse, purchaseRequestContent, response.Value.Properties.ReservationOrderId);
+        }
+
+        [TestCase]
+        [RecordedTest]
+        public async Task TestPurchaseReservationOrderSharedScopeUpfront()
+        {
+            var purchaseRequestContent = CreatePurchaseRequestContent("Shared", "Upfront");
+            var response = await Tenant.CalculateReservationOrderAsync(purchaseRequestContent);
+            var purchaseResponse = await collection.CreateOrUpdateAsync(WaitUntil.Completed, response.Value.Properties.ReservationOrderId, purchaseRequestContent);
+
+            await purchaseResponse.WaitForCompletionAsync();
+            validateTestResult(purchaseResponse, purchaseRequestContent, response.Value.Properties.ReservationOrderId);
+        }
+
+        [TestCase]
+        [RecordedTest]
+        public async Task TestPurchaseReservationOrderSingleScopeMonthly()
+        {
+            var purchaseRequestContent = CreatePurchaseRequestContent("Single", "Monthly");
+            var response = await Tenant.CalculateReservationOrderAsync(purchaseRequestContent);
+            var purchaseResponse = await collection.CreateOrUpdateAsync(WaitUntil.Completed, response.Value.Properties.ReservationOrderId, purchaseRequestContent);
+            await purchaseResponse.WaitForCompletionAsync();
+            validateTestResult(purchaseResponse, purchaseRequestContent, response.Value.Properties.ReservationOrderId);
+        }
+
+        [TestCase]
+        [RecordedTest]
+        public async Task TestPurchaseReservationOrderSingleScopeUpfront()
+        {
+            var purchaseRequestContent = CreatePurchaseRequestContent("Single", "Upfront");
+            var response = await Tenant.CalculateReservationOrderAsync(purchaseRequestContent);
+            var purchaseResponse = await collection.CreateOrUpdateAsync(WaitUntil.Completed, response.Value.Properties.ReservationOrderId, purchaseRequestContent);
+            await purchaseResponse.WaitForCompletionAsync();
+            validateTestResult(purchaseResponse, purchaseRequestContent, response.Value.Properties.ReservationOrderId);
+        }
+
+        private void validateTestResult(ArmOperation<ReservationOrderResponseResource> purchaseResponse, PurchaseRequestContent purchaseRequest, string reservationOrderId)
```

Comment:
Maybe a renaming is better:
```suggestion
        private void TestCreatePurchaseResponse(ArmOperation<ReservationOrderResponseResource> purchaseResponse, PurchaseRequestContent purchaseRequest, string reservationOrderId)
```